README.md

```diff
@@ -1,28 +1,71 @@
+
+Once the scripts finishes you can access the jenkins instance from the browser at ```http://localhost:8080```
+
+### Cloud provision
+
+There are additional requirements, beside having loaded OpenStack credentials, for the cloud provision, in this case the setup is done in a cloud instance and you should be able to spin it up, the needed packages can be installed in Ubuntu with:
+
+    $ sudo apt-get install cloud-utils python-novaclient
+
+To setup the CI instance in the cloud just execute:
+
+    $ ./bin/cloud-provision.sh <cloud_credentials_path>
+
+This command creates a new VM with two containers in it, as described in this image:
+
+![Block Diagram](/img/snappy-jenkins.png?raw=true)
+
+* Reverse proxy instance: it has a Nginx process listening on port 8081 that forwards request to path ```/ghprbhook``` to the jenkins instance
+
+* Jenkins instance: the same as in the local provision case.
 
-bin/local-provision.sh creates the container locally and shares jenkins home in /tmp/jenkins. Once it finish the app should be accesible from http://localhost:8080.
+It also setups a security group that allows access to port 8081 from everywhere and ports 8080 and 22 from a local range (by default 10.0.0.0/8). All this configuration is done this way in order to facilitate a secure connection from the GitHub webhook, as detailed in the next section.
 
-bin/cloud-provision.sh uses the credentials of the current user to set up a cloud instance that will host the jenkins container.
+## GitHub integration
 
-Both scripts require as a parameter the path to the directory where the openstack credentials are stored. This credentials are used by jenkins to spin up the instances where the tests are run. In the case of cloud provision you can use different credentials for launching the jenkins host (this would be the credentials loaded for the user executing the script) and for jenkins to use while executing the snappy tests.
+The Jenkins container has installed the GitHub Pull Request Builder Plugin [3] to allow the triggering of jobs in response to events from GitHub. The default configuration in the container leaves almost all setup, there's only a few things left to do:
 
-They can be executed like this:
+* First of all, you should assign a floating IP to your VM instance so that in can be reached from GitHub. The container infrastructure and security group assigned make it secure to expose it to the wild, as explained earlier.
 
-    $ ./bin/cloud-provision.sh ~/.openstack
+* The ```github-snappy-integration-tests``` job is configured to receive payloads from GitHub in response to events. It points to the ubuntu-core/snappy repository, you can change this to access one of your own repos to try the hook.
```

Comment:
IMO we should trigger all them from a single one. we can talk about that later,  I'll rename this to align with the changes in the other branch